src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -315,4 +328,51 @@ private void saveWithDifferentEncoding(Path file, boolean selectedOnly, Charset
             }
         }
     }
+
+    /**
+     * Handle JabRef git integration action. This method is called when the user save the database.
+     */
+    public void automaticGitUpdate(Path filePath) throws IOException, GitAPIException {
+        GitHandler git = new GitHandler(filePath.getParent(), preferences.getGitPreferences());
+        List<GitChange> changes = new ArrayList<GitChange>();
+        BibGitContext bibGitContext = new BibGitContext();
+        if (preferences.getGitPreferences().getAutoCommit()) {
+            String automaticCommitMsg = "Automatic update via JabRef";
+            git.createCommitWithSingleFileOnCurrentBranch(filePath.getFileName().toString(), automaticCommitMsg);
+        }
+        if (preferences.getGitPreferences().getAutoSync()) {
+            try {
+                git.pullOnCurrentBranch();
+            } catch (IOException ex1) {
+                if (ex1.getMessage().equals("No git credentials")) {
```

Comment:
Doesn't git have a separeate exception? If not, please add a comment on this line of code.

(I would have assumed that the git exeptions do some inhertiance)